### PR TITLE
Refactor: Extract MovieService from MediaManager

### PR DESCRIPTION
**Pull request by kasuboski**
## Summary

Extracts a `MovieService` from the `MediaManager` god object as part of the incremental service decomposition in #139.

`MediaManager` retains thin delegation methods for backward compatibility. All callers continue to work without modification.

## Scope

**Moved to `MovieService`** (CRUD + listing + search/detail):
- `AddMovieToLibrary`, `DeleteMovie`, `UpdateMovieMonitored`, `UpdateMovieQualityProfile`
- `ListMoviesInLibrary`
- `SearchMovie`, `GetMovieDetailByTMDBID`

**Intentionally NOT moved** (extracted in future PRs):
- Indexing (`IndexMovieLibrary`) → `IndexerService`
- Reconciliation (`movie_reconcile.go`) → cross-cutting deps
- Metadata (`GetMovieMetadata`, `UpdateMovieMetadataFromTMDB`, etc.) → `MetadataService`

## Design

- `MovieMetadataProvider` interface (2 methods) decouples `MovieService` from the metadata subsystem
- `MediaManager` implements the interface implicitly
- Package-level helpers (`initialMovieState`, `parseMediaResult`, `filterAndMap`) remain shared
- Pointer receivers on all `MovieService` methods

## Files Changed

| File | Change |
|------|--------|
| `pkg/manager/movie_service.go` | **NEW** — `MovieService` with 7 public methods |
| `pkg/manager/movie_service_test.go` | **NEW** — 26 unit tests |
| `pkg/manager/manager.go` | Removed ~200 lines, added `movieService` field + 7 delegation wrappers |
| `pkg/manager/manager_test.go` | Removed 4 movie test functions (now on `MovieService`) |
| `pkg/manager/movie_reconcile_test.go` | Sync tmdb client to `movieService` in 5 test cases |

## Testing

- 26 unit tests in `movie_service_test.go` covering all 7 methods (happy paths, error paths, edge cases)
- `go test ./...` passes
- `go vet ./...` clean

#139 (partial — covers `MovieService` extraction; remaining services tracked in the issue)

## Summary by CodeRabbit

* **Refactor**
  * Movie-related logic reorganized into a dedicated movie service for more consistent searching, details, adding, updating, listing, and deleting behavior.

* **Tests**
  * Test suite reworked: removed older manager-level movie tests and added focused unit tests for the new movie service to improve coverage and reliability.

* **Style**
  * Minor comment corrections for clarity.

### CODE REVIEW

File: pkg/manager/movie_service.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	}
+
+	log.Debug("created movie", zap.Any("movie", movie))
+
+	movie, err = s.movieStorage.GetMovie(ctx, id)
+	if err != nil {
+		log.Warn("failed to get created movie", zap.Error(err))
+	}
+
+	return movie, nil
```
[the remark this answers is not in the document]
Comment (kasuboski) [reply pkg/manager/movie_service.go:105]: @copilot fix it

File: pkg/manager/movie_service_test.go [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+			MovieMetadataID:  ptr.To(int32(1)),
+			Path:             ptr.To("Test Movie"),
+		},
+		State: storage.MovieStateMissing,
+	}, nil)
+
+	movie, err := svc.AddMovieToLibrary(ctx, AddMovieRequest{TMDBID: 1234, QualityProfileID: 1})
+	require.NoError(t, err)
+	require.NotNil(t, movie)
+	assert.Equal(t, int32(1), movie.ID)
```
Comment (kdwils) [inline pkg/manager/movie_service_test.go]: we should probably do entire struct comparisons instead of individual fields

File: pkg/manager/movie_service.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+		metadataProvider: metadataProvider,
+	}
+}
+
+// ---------------------------------------------------------------------------
+// CRUD Operations
+// ---------------------------------------------------------------------------
+
+// AddMovieToLibrary adds a movie to be managed by mediaz.
+// TODO: check status of movie before doing anything else.. do we already have it tracked? is it downloaded or already discovered? error state?
```
Comment (kdwils) [inline pkg/manager/movie_service.go:52]: is this something we still need to do? should make an issue for it if so

**Review by kdwils — APPROVED** [review]
